tests/Tests/500__jupyterhub/custom-image.robot

```diff
@@ -1,51 +1,135 @@
-    Begin Web Test
+    Verify Image Name  ${IMG_NAME}
+    Verify Image Description  ${IMG_NAME}  ${IMG_DESCRIPTION}
+    Verify User  ${IMG_NAME}  ${TEST_USER.USERNAME}
+    Launch JupyterHub Spawner From Dashboard
+
+    # These keywords need to be reworked to function here
+    #${spawner_description}=  Fetch Image Tooltip Description  ${IMAGESTREAM_NAME}
+    #${spawner_packages}=  Fetch Image Tooltip Info  ${IMAGESTREAM_NAME}
+    #${spawner_software}=  Fetch Image Description Info  ${IMAGESTREAM_NAME}
+    #Should Match  ${spawner_description}  ${IMG_DESCRIPTION}
+    #Should Match  ${spawner_software}  ${IMG_SOFTWARE}
+    #Should Match  ${spawner_packages}  ${IMG_PACKAGES}
+
+    Spawn Notebook With Arguments  image=${IMAGESTREAM_NAME}  size=Default
+    [Teardown]  Custom Image Teardown
+
+Test Duplicate Image
+    [Documentation]  Test adding two images with the same name (should fail)
+    Apply Custom ImageStream And Check Status
+    Sleep  1
+    Import New Image    ${IMG_URL}    ${IMG_NAME}    ${IMG_DESCRIPTION}
+    ...    software=${IMG_SOFTWARE}
+    ...    packages=${IMG_PACKAGES}
+    RHODS Notification Drawer Should Contain  Unable to add notebook image ${IMG_NAME}
+    Sleep  1
+    Delete Image  ${IMG_NAME}
+    Reset Image Name
+
+Test Bad Image URL
+    [Documentation]  Test adding an image with a bad repo URL (should fail)
+    ${OG_URL}=  Set Variable  ${IMG_URL}
+    ${IMG_URL}=  Set Variable  quay.io/RandomName/RandomImage:v1.2.3
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Apply Custom ImageStream And Check Status
+    RHODS Notification Drawer Should Contain  Unable to add notebook image ${IMG_NAME}
+    ${IMG_URL}=  Set Variable  ${OG_URL}
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Reset Image Name
+    # No image created, teardown not needed
+
+Test Bad Image Import
+    [Documentation]  Import a broken image and confirm it cannot spawn
+    ${OG_URL}=  Set Variable  ${IMG_URL}
+    ${IMG_URL}=  Set Variable  randomstring
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Apply Custom ImageStream And Check Status
+    Get ImageStream Metadata And Check Name
     Launch JupyterHub Spawner From Dashboard
-    Spawn Notebook With Arguments  image=${IMG_NAME}  size=Default
+    Run Keyword And Continue On Failure  Spawn Notebook With Arguments  image=${IMAGESTREAM_NAME}  size=Default
+    ${IMG_URL}=  Set Variable  ${OG_URL}
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Reset Image Name
 
 
 *** Keywords ***
 Custom Image Teardown
     [Documentation]    Closes the JL server and deletes the ImageStream
-    End Web Test
-    OpenShiftCLI.Delete    kind=ImageStream    field_selector=metadata.name==${IMG_NAME}
-    ...    namespace=redhat-ods-applications
+    Clean Up Server
+    Stop JupyterLab Notebook Server
+    Go To  ${ODH_DASHBOARD_URL}
+    Open Notebook Images Page
+    Sleep  1
+    Delete Image  ${IMG_NAME}
+    Reset Image Name
 
 Apply Custom ImageStream And Check Status
```

Comment:
just a naming suggestion, maybe `Apply Custom ImageStream` sounds like the operation is done using OC. A better name could be `Create Custom Image And Check Status`